server/sonar-server/src/main/java/org/sonar/server/computation/metric/Metric.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.metric;
+
+import org.sonar.server.computation.measure.Measure;
+
+public interface Metric {
+  /**
+   * The Metric's key is its domain identifier.
+   */
+  String getKey();
+
+  String getName();
+
+  MetricType getMetricType();
+
+  enum MetricType {
+    INT(Measure.ValueType.INT),
+    MILLISEC(Measure.ValueType.INT),
+    RATING(Measure.ValueType.INT),
+    WORK_DUR(Measure.ValueType.LONG),
+    FLOAT(Measure.ValueType.DOUBLE),
+    PERCENT(Measure.ValueType.DOUBLE),
+    BOOL(Measure.ValueType.BOOLEAN),
+    STRING(Measure.ValueType.STRING),
+    DISTRIB(Measure.ValueType.STRING),
+    DATA(Measure.ValueType.STRING),
+    LEVEL(Measure.ValueType.NO_VALUE);
```

Comment:
see design of Measure (the legacy and the new one)
"level" is not stored as a the value of the Measure but in a specific field (which all measures, whichever their type may have populated)
hence, for consistency, the Measure has no value but an always populated QualityGateStatus

Please note that the LEVEL is badly named or at least misleading (its really a metric of type QualityGate), but this is the value stored in DB (and also exposed by WebService I reckon)